pkg/reconciler/ingress/ingress.go

```diff
@@ -232,8 +232,16 @@ func (r *Reconciler) reconcileIngress(ctx context.Context, ing *v1alpha1.Ingress
 	}
 
 	if ready {
-		publicLbs := getLBStatus(publicGatewayServiceURLFromContext(ctx))
-		privateLbs := getLBStatus(privateGatewayServiceURLFromContext(ctx))
+		publicLbs, err := getLBStatuses(ctx, v1alpha1.IngressVisibilityExternalIP)
+		if err != nil {
+			return fmt.Errorf("failed to get external LB status: %w", err)
+		}
+
+		privateLbs, err := getLBStatuses(ctx, v1alpha1.IngressVisibilityClusterLocal)
```

Comment:
In my understanding, this will set multiple gateways to the LB status when users configured multiple gateways. Did that work with serving? For example, here is the code which disallows it - 
https://github.com/knative/serving/blob/9b0d1851055e91ced3965dc3cb8d9edb08f833ee/pkg/reconciler/route/resources/service.go#L87-L91

Reply:
Ok, I wasn't aware of such limitation at serving side.

Small remark tho, we can have multiple istio gateways that point to the same istio ingress gateway (and therefore LB)